$MirOS: ports/shells/zsh/patches/patch-Src_builtin_c,v 1.1 2009/09/06 16:31:12 tg Exp $

	Octets and Multibyte Characters are handled the same
	in zsh on Debian, so we must do so in MirBSD as well

--- Src/builtin.c.orig	Mon Oct 18 09:31:12 2010
+++ Src/builtin.c	Tue Mar  8 17:18:23 2011
@@ -4291,6 +4291,10 @@ bin_print(char *name, char **args, Optio
 			mb_metacharinit();
 			(void)mb_metacharlenconv(metafy(curarg+1, curlen-1,
 							META_USEHEAP), &cc);
+#ifdef iswoctet
+			if (iswoctet(cc))
+				cc &= 0xFF;
+#endif
 		    }
 		    else
 			cc = WEOF;
